src/ripple/ledger/impl/RawStateTable.cpp

```diff
@@ -24,6 +24,133 @@
 namespace ripple {
 namespace detail {
 
+class RawStateTable::sles_iter_impl
+    : public ReadView::sles_type::iter_base
+{
+private:
+    std::shared_ptr<SLE const> sle0_;
+    ReadView::sles_type::iterator iter0_;
+    ReadView::sles_type::iterator end0_;
+    std::shared_ptr<SLE const> sle1_;
+    items_t::const_iterator iter1_;
+    items_t::const_iterator end1_;
+
+public:
+    sles_iter_impl (sles_iter_impl const&) = default;
+
+    sles_iter_impl (items_t::const_iterator iter1,
+        items_t::const_iterator end1,
+            ReadView::sles_type::iterator iter0,
+                ReadView::sles_type::iterator base0)
+        : iter0_ (iter0)
+        , end0_ (base0)
+        , iter1_ (iter1)
+        , end1_ (end1)
+    {
+        if (iter0_ != end0_)
+            sle0_ = *iter0_;
+        if (iter1_ != end1)
+        {
+            sle1_ = iter1_->second.second;
+            skip ();
+        }
+    }
+
+    std::unique_ptr<base_type>
+    copy() const override
+    {
+        return std::make_unique<sles_iter_impl>(*this);
+    }
+
+    bool
+    equal (base_type const& impl) const override
+    {
+        auto const& other = dynamic_cast<
+            sles_iter_impl const&>(impl);
```

Comment:
Perfect. I don't use `dynamic_cast` enough to commit the behaviors to memory. Thanks.